client/src/ui/ToasterWebview.ts

```diff
@@ -0,0 +1,202 @@
+import { BitbakeDriver } from '../driver/BitbakeDriver'
+import { pty } from '../utils/ProcessUtils'
+import { logger } from '../lib/src/utils/OutputLogger'
+import { loadBitbakeSettings } from '../lib/src/BitbakeSettings'
+
+const DEFAULT_TOASTER_PORT = 8000
+
+export class ToasterPanel {
+  /**
+  * Track the currently panel. Only allow a single panel to exist at a time.
+  */
+  public static currentPanel: ToasterPanel | undefined
+  public static toasterProcess: IPty | undefined
+  public static readonly viewType = 'bitbake.toasterView'
+
+  private readonly _panel: vscode.WebviewPanel
+  private readonly _disposables: vscode.Disposable[] = []
+
+  public static createOrShow (): void {
+    const column = vscode.window.activeTextEditor !== undefined
+      ? vscode.window.activeTextEditor.viewColumn
+      : undefined
+
+    // If we already have a panel, show it.
+    if (ToasterPanel.currentPanel !== undefined) {
+      ToasterPanel.currentPanel._panel.reveal(column)
+      return
+    }
+
+    // Otherwise, create a new panel.
+    const panel = vscode.window.createWebviewPanel(
+      ToasterPanel.viewType,
+      'Yocto Toaster',
+      column ?? vscode.ViewColumn.One,
+      getWebviewOptions()
+    )
+
+    ToasterPanel.currentPanel = new ToasterPanel(panel)
+  }
+
+  public static async startToasterInBrowser (): Promise<void> {
+    await ToasterPanel.runToasterCommand('source toaster start', true)
+    const url = `http://localhost:${DEFAULT_TOASTER_PORT}`
+    await vscode.env.openExternal(vscode.Uri.parse(url)).then(success => {
+      if (!success) {
+        void vscode.window.showErrorMessage(`Failed to open URL ${url}`)
+      }
+    })
+  }
+
+  // Close the webview and stop the toaster
+  public static stop (): void {
+    ToasterPanel.currentPanel?.dispose()
+  }
+
+  // A custom replicate of runBitbakeCommand() from BitbakeTerminal.ts
+  private static async runToasterCommand (command: string, isBackground: boolean = false, terminalName: string = 'Toaster'): Promise<IPty> {
+    let terminal: BitbakeTerminal | undefined
+    for (const t of bitbakeTerminals) {
+      if (!t.pty.isBusy()) {
+        terminal = t
+        terminal.pty.changeNameEmitter.fire(terminalName)
+        break
+      }
+    }
+
+    if (terminal === undefined) {
+      const bitbakeDriver = new BitbakeDriver()
+      terminal = new BitbakeTerminal(terminalName, bitbakeDriver)
+
+      await new Promise(resolve => terminal?.pty.onDidOpen.event(resolve))
+    }
+
+    if (!isBackground) {
+      terminal.terminal.show()
+    }
+    const process = ToasterPanel.spawnToasterProcess(command)
+    await terminal.pty.runProcess(process, command, terminalName)
+    return await process
+  }
+
+  // A custom replicate of spawnBitbakeProcess() from BitbakeDriver.ts
+  private static async spawnToasterProcess (command: string): Promise<IPty> {
+    // TODO: check if the env file exists
+    const script = 'source oe-init-build-env' + ' && ' + command
```

Comment:
This does not respect the user settings. Also, all commands should be generated by BitbakeDriver. It's that class' responsibility and allow us to quickly edit all commands at a glance if we add more settings in the future.